src/handlers/decision.rs

```diff
@@ -152,6 +162,119 @@ pub(super) async fn handle_command(
     }
 }
 
+fn build_status_comment(
+    history: &BTreeMap<String, Vec<UserStatus>>,
+    current: &BTreeMap<String, Option<UserStatus>>,
+) -> anyhow::Result<String> {
+    let mut comment = "| Team member | State |\n|-------------|-------|".to_owned();
+    for (user, statuses) in history {
+        let mut user_statuses = format!("\n| {} |", user);
+
+        // previous stasuses
+        for status in statuses {
+            let status_item = format!(" ~~{}~~ ", resolution_to_str(&status.resolution));
+            user_statuses.push_str(&status_item);
+        }
+
+        // current status
+        let current_status = current.get(user).unwrap();
+        let user_resolution;
+        match current_status {
+            Some(status) => user_resolution = resolution_to_str(&status.resolution),
+            _ => user_resolution = "".to_string(),
+        }
+        let status_item = format!(" **{}** |", user_resolution);
+        user_statuses.push_str(&status_item);
+
+        comment.push_str(&user_statuses);
+    }
+
+    println!("{}", comment);
+    Ok(comment)
+}
+
+fn resolution_to_str(resolution: &Resolution) -> String {
+    match resolution {
+        Merge => "merge".to_owned(),
+        Hold => "hold".to_owned(),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
```

Comment:
Could be good to add a test for the case where something returns an error? For example the user names are inconsistent between `history` and `current` or something like that

Reply:
Definitely!